packages/flutter/lib/src/foundation/binding.dart

```diff
@@ -295,8 +299,11 @@ abstract class BindingBase {
     registerServiceExtension(
       name: name,
       callback: (Map<String, String> parameters) async {
-        if (parameters.containsKey('enabled'))
-          await setter(parameters['enabled'] == 'true');
+        if (parameters.containsKey('enabled')) {
+          final bool value = parameters['enabled'] == 'true';
+          await setter(value);
+          _postExtensionToggledEvent(name, value);
```

Comment:
nit: change this to 
_postExtensionChangedEvent
as not all the states are truly just toggled.